Samples/src/IntroToIQSharp/Notebook.ipynb

```diff
@@ -56,138 +90,425 @@
+    "        let r = M(q);\n",
+    "        \n",
+    "        Reset(q);\n",
+    "        return r;\n",
+    "    }\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
    "metadata": {},
    "outputs": [
     {
      "data": {
-      "application/json": "[\"HelloAndres\"]",
+      "application/json": "1",
+      "text/plain": [
+       "One"
+      ]
+     },
+     "execution_count": 5,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "%simulate QNRG"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "notes"
+    }
+   },
+   "source": [
+    "As mentioned, one of the key characteristics of a qubit is that during execution it's state represents the probabiliy of measuring $|0\\rangle$ or $|1\\rangle$. In particular, the [`H`](https://docs.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.primitive.h) primitive operation modifies the state of a qubit from $|0\\rangle$ to a state in which there is 50% probability of measuring  $|0\\rangle$ and 50% probability of measuring  $|1\\rangle$.\n",
+    "\n",
+    "> When the state of a qubit is such that it is not a 100% $|0\\rangle$ or 100% $|1\\rangle$, \n",
+    "> we say the qubit is in **superposition**."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "`QNRG` demonstrates some basic Q# features:\n",
+    "\n",
+    "- Immutable variables are created using `let`, in the previous example, the variable `r` is assigned the value returned by invoking `M` on qubit `q`.\n",
+    "- When de-allocated, qubits must be explicitly set back to $|0\\rangle$, otherwise the simulator will report a runtime error. An easy way to achieve this is invoking [`Reset`](https://docs.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.primitive.reset).\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "notes"
+    }
+   },
+   "source": [
+    "\n",
+    "To verify the behavior of `QNRG`, we can create another Q# operation that invokes it 10 times and prints the result to the console:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "outputs": [
+    {
+     "data": {
+      "application/json": "[\"Main\"]",
       "text/html": [
-       "<ul><li>HelloAndres</li></ul>"
+       "<ul><li>Main</li></ul>"
       ],
       "text/plain": [
-       "HelloAndres"
+       "Main"
       ]
      },
-     "execution_count": 3,
+     "execution_count": 6,
      "metadata": {},
      "output_type": "execute_result"
     }
    ],
    "source": [
-    "operation HelloAndres() : Unit {\n",
-    "    HelloQ(\"Andres\");\n",
+    "operation Main() : Unit {\n",
+    "    for(i in 1..10) {\n",
+    "        Message($\"{i}: {QNRG()}\");\n",
```

Comment:
Do we want to quickly note that `$` strings work a lot like C#'s `$` strings and Python's `f` strings?